website/views.py

```diff
@@ -245,12 +264,28 @@ def part(part_number: int) -> ResponseReturnValue:
     if part.last_modified is not None:
         part.last_modified = part.last_modified.replace(tzinfo=UTC)
 
+    # Get top-level comments
+    comments = (
+        db.session.scalars(
+            select(Comment)
+            .where(Comment.part_id == part_number, Comment.parent_id == None)
+            .order_by(Comment.date.desc())
+            .options(
+                joinedload(Comment.author),
+                joinedload(Comment.replies).joinedload(Comment.author),
+            )
+        )
+        .unique()
+        .all()
+    )
```

Comment:
The comments get loaded with `datetime`s without timezone, which results in wrong dates. For _last modified_ this is worked around by replacing TZ, but maybe something like a [custom type](https://docs.sqlalchemy.org/en/20/core/custom_types.html#store-timezone-aware-timestamps-as-timezone-naive-utc) would be better.

https://github.com/Indystrycc/OpenRoboticPlatform-website/blob/28f742c6de0a7cd0938c3e1d3bd54f196f516a13/website/views.py#L244-L246